client/components/progress-bar/index.jsx

```diff
@@ -54,6 +58,14 @@ export default class ProgressBar extends PureComponent {
 		);
 	}
 
+	componentWillReceiveProps( nextProps ) {
+		this.setState( { allTimeMax: Math.max( this.state.allTimeMax, nextProps.value ) } );
+	}
+
+	shouldComponentUpdate( nextProps, nextState ) {
+		return this.state.allTimeMax < nextState.allTimeMax;
+	}
```

Comment:
I don't think we want this here. Note how `render()` depends on `this.props.className` - if that changes the component won't update.

This one component shouldn't be a problem for performance even if some extra renders come through - it's just a couple DOM elements and nothing else is really happening while the progress bar updates.

How would you feel about simply omitting `shouldComponentUpdate`?

Reply:
Why you don't want it there?

Isn't `shouldComponentUpdate` working correctly?